target/linux/ath79/dts/qca9531_compex_wpj531.dts

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca953x.dtsi"
+
+/ {
+	compatible = "compex,wpj531", "qca,qca9531";
+	model = "Compex WPJ531 (16MB flash)";
+
+	chosen {
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		sig1 {
+			label = "wpj531:red:sig1";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+		};
+
+		sig2 {
+			label = "wpj531:yellow:sig2";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+
+		sig3 {
+			label = "wpj531:green:sig3";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+
+		sig4 {
+			label = "wpj531:green:sig4";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+		poll-interval = <30>;
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&uart {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			uboot:	partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x030000>;
```

Comment:
Fix size and add uboot-env as discussed.

Reply:
i'm reluctant to label the section as u-boot-env because it doesn't look like a u-boot environment to me (no names, separators).
it appears there are two distinct config sections.

``` 
$ hexdump -C /dev/mtd0
...
0000f400  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
*
0002e000  20 13 04 17 00 00 00 00  00 00 00 00 00 00 00 00  | ...............|
0002e010  04 f0 21 38 71 cb 8e fd  04 f0 21 38 71 cc 4a cf  |..!8q.....!8q.J.|
0002e020  04 f0 21 38 71 cd f3 b5  04 f0 21 38 71 ce 3f 4c  |..!8q.....!8q.?L|
0002e030  00 00 00 00 00 00 55 5f  00 00 00 00 00 00 90 84  |......U_........|
0002e040  00 00 00 00 00 00 5a c6  00 00 00 00 00 00 b1 d4  |......Z.........|
0002e050  11 f6 06 93 11 f6 06 93  00 00 00 03 00 00 00 00  |................|
0002e060  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
0002e0f0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
*
0002e800  0f f9 99 dc a8 ed 27 aa  4f 5d 08 ac d3 18 71 02  |......'.O]....q.|
0002e810  d4 1f 1d 0f 14 af d2 84  52 9d ac d9 02 2b 24 6f  |........R....+$o|
0002e820  94 90 d6 bb e7 a2 d1 c9  c4 19 6f 88 e6 fe aa b4  |..........o.....|
0002e830  a1 51 75 d5 5d 2a a2 f3  b7 55 49 b6 52 48 4b ec  |.Qu.]*...UI.RHK.|
0002e840  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
*
00030000
```

Reply:
Please re-read my earlier comment on the partition sizes!

Reply:
https://github.com/openwrt/openwrt/pull/2265#discussion_r308260879

You can BTW make your life easier by not marking everything as resolved ...

Reply:
earlier you said something about
`mtdparts=ath-nor0:256k(u-boot),64k(u-boot-env),6336k(rootfs),1408k(uImage),8256k(mib0),64k(ART)`
That would mean u-boot-env was from 0x40000 to 0x50000, which is overwritten now.
Still wonder about the four MAC addresses.

Reply:
i'm sorry for having resolved the conversion.
i didn't fully grasp the difference between our positions.

my response is here:
https://github.com/openwrt/openwrt/pull/2265#discussion_r308955137

should i resolve this one to have only active discussion about the topic?

Reply:
I would rather leave it open, so we see that there still might be something to do here ...

Reply:
You could try to execute saveenv from uboot and see, where the environment is written to.